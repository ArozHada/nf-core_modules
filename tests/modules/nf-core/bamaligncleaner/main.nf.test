nextflow_process {

    name "Test Process BAMALIGNCLEANER"
    script "modules/nf-core/bamaligncleaner/main.nf"
    process "BAMALIGNCLEANER"
    config "tests/modules/nf-core/bamaligncleaner/nextflow.config"
    tag "bamaligncleaner"

    test("Sarscov2 Illumina SE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                             file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true) 
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
