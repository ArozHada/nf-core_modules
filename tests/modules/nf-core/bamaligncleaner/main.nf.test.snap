{
    "Sarscov2 Illumina SE Sorted BAM": {
        "content": [
            [
                "test.bam:md5,173cdb4c2713b77c528cac36ca2610fb",
                "versions.yml:md5,b66f7ae54bcd8686660afdd384864d40"
            ]
        ],
        "timestamp": "2023-04-07T23:17:47+0000"
    }
}